apps/client/src/app/components/admin-market-data/create-asset-profile-dialog/create-asset-profile-dialog.component.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Inject,
+  OnDestroy,
+  OnInit
+} from '@angular/core';
+import {
+  AbstractControl,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { AdminMarketDataItem } from '@ghostfolio/common/interfaces';
+import { CreateAssetProfileDialogParams } from '@ghostfolio/client/components/admin-market-data/create-asset-profile-dialog/interfaces/interfaces';
+import { AdminService } from '@ghostfolio/client/services/admin.service';
+
+@Component({
+  host: { class: 'h-100' },
+  selector: 'gf-create-asset-profile-dialog',
+  templateUrl: 'create-asset-profile-dialog.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CreateAssetProfileDialog implements OnInit, OnDestroy {
+  public createAssetProfileForm: FormGroup;
+
+  constructor(
+    public readonly adminService: AdminService,
+    public readonly changeDetectorRef: ChangeDetectorRef,
+    @Inject(MAT_DIALOG_DATA) public data: CreateAssetProfileDialogParams,
+    public readonly dialogRef: MatDialogRef<CreateAssetProfileDialog>,
+    public readonly formBuilder: FormBuilder
+  ) {}
+
+  ngOnInit() {
+    this.createAssetProfileForm = this.formBuilder.group({
+      searchSymbol: new FormControl(null, [
+        Validators.required,
+        this.validateSymbolAlreadyExists(this.data.existingSymbols)
+      ])
+    });
+  }
+
+  ngOnDestroy(): void {}
+
+  onSubmit() {
+    this.dialogRef.close({
+      dataSource:
+        this.createAssetProfileForm.controls['searchSymbol'].value.dataSource,
+      symbol: this.createAssetProfileForm.controls['searchSymbol'].value.symbol
+    });
+  }
+
+  onCancel() {
+    this.dialogRef.close();
+  }
+
+  validateSymbolAlreadyExists(
```

Comment:
I would not validate in the frontend. In the future, the client will no longer have the entire data at its disposal.